StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1120CodeFixProvider.cs

```diff
@@ -113,12 +96,12 @@ private static bool TriviaHasLeadingContentOnLine(SyntaxNode root, IReadOnlyList
             return false;
         }
 
-        private static bool TriviaHasTrailingContentOnLine(SyntaxNode root, IReadOnlyList<SyntaxTrivia> triviaList)
+        private static bool TriviaHasTrailingContentOnLine(SyntaxNode root, SyntaxTrivia commentTrivia)
         {
-            var nodeAfterTriviaStart = triviaList[triviaList.Count - 1].SpanStart - 1;
+            var nodeAfterTriviaStart = commentTrivia.Span.End + 1;
             var nodeAfterTrivia = root.FindNode(new Microsoft.CodeAnalysis.Text.TextSpan(nodeAfterTriviaStart, 1));
 
-            if (nodeAfterTrivia.GetLineSpan().StartLinePosition.Line == triviaList[triviaList.Count - 1].GetLineSpan().EndLinePosition.Line)
+            if (nodeAfterTrivia.GetLineSpan().StartLinePosition.Line == commentTrivia.GetLineSpan().EndLinePosition.Line && !nodeAfterTrivia.GetTrailingTrivia().Contains(commentTrivia))
```

Comment:
The if and return statements below can be replaced by a single return statement.